stellargraph/core/graph.py

```diff
@@ -778,20 +937,122 @@ def create_graph_schema(self, create_type_maps=True, nodes=None):
                         node_types[node_type_map[edge[1]]],
                     )
                 )
-                for edge in self.edges(keys=True, data=True)
+                for edge in self._graph.edges(keys=True, data=True)
             }
 
             gs.node_type_map = node_type_map
             gs.edge_type_map = edge_type_map
 
         return gs
 
+    ######################################################################
+    # Generic graph interface:
+
+    def is_directed(self) -> bool:
+        return self._graph.is_directed()
+
+    def number_of_nodes(self) -> int:
+        return self._graph.number_of_nodes()
+
+    def number_of_edges(self) -> int:
+        return self._graph.number_of_edges()
+
+    def nodes(self) -> Iterable[Any]:
+        return self._graph.nodes()
 
-class StellarGraph(StellarGraphBase, MultiGraph):
-    def __init__(self, incoming_graph_data=None, **attr):
-        super().__init__(incoming_graph_data, **attr)
+    def edges(self) -> Iterable[Any]:
+        return self._graph.edges()
 
+    def has_node(self, node: Any) -> bool:
+        return self._graph.__contains__(node)
 
-class StellarDiGraph(StellarGraphBase, MultiDiGraph):
-    def __init__(self, incoming_graph_data=None, **attr):
-        super().__init__(incoming_graph_data, **attr)
+    def neighbour_nodes(self, node: Any) -> Iterable[Any]:
+        if self.is_directed():
+            in_nodes = {e[0] for e in self._graph.in_edges(node)}
+            out_nodes = {e[1] for e in self._graph.out_edges(node)}
+            return in_nodes | out_nodes
+        return nx.neighbors(self._graph, node)
+
+    def in_nodes(self, node: Any) -> Iterable[Any]:
+        if self.is_directed():
+            return {e[0] for e in self._graph.in_edges(node)}
+        return nx.neighbors(self._graph, node)
+
+    def out_nodes(self, node: Any) -> Iterable[Any]:
+        if self.is_directed():
+            return {e[1] for e in self._graph.out_edges(node)}
+        return nx.neighbors(self._graph, node)
+
+    ########################################################################
+    # Heavy duty methods:
+
+    def node_degrees(self) -> Mapping[Any, int]:
+        return self._graph.degree()
+
+    def adjacency_weights(self):
```

Comment:
There is a conceptual problem here in that this ticket is (supposedly) about making minimal changes to stop StellarGraph inheriting from NetworkX, whereas the subsequent ticket (see #520) is about distilling the StellarGraph interface and making a time- and memory-efficient implementation that does not use NetworkX, including encapsulating edge and node data in a standard way. So some of the concerns, and some of my answers, keep crossing between these two tickets.

This is one of the reasons I've been trying to push ahead with crystallising what the StellarGraph interface currently supports (i.e. everything that NetworkXStellarGraph can do, rightly or wrongly, including `create_graph_schema()`), in advance of trying to optimise what it actually should do. And I wholeheartedly agree that this approach runs the risk of implementing a suboptimal design, but we're working in the dark otherwise. We all need a good think about this.

Reply:
I agree with your points, but as I understand it this function was introduced to the class in this ticket, or am I mistaken?

Reply:
The actual function `adjacency_weights()` was created in this ticket, but the functionality itself was already in use in several places. This change just hid away the fact that the code was using NetworkX-specific structures and calls.

Reply:
OK, so can we can change the name to `to_adjacency_matrix`?